src/terms/local.jl

```diff
@@ -76,13 +76,31 @@ function (::AtomicLocal)(basis::PlaneWaveBasis{T}) where {T}
     # positions, this involves a form factor (`local_potential_fourier`)
     # and a structure factor e^{-i G·r}
 
-    pot_fourier = map(G_vectors(basis)) do G
-        pot = sum(model.atom_groups) do group
+    Gvecs = G_vectors(basis)
+    qnorms = map(G -> norm(model.recip_lattice * G), Gvecs)
+    uqnorms = unique(qnorms)
+    iuq2iq = map(iuq -> findall(qnorm -> isequal(qnorm, uqnorms[iuq]), qnorms), 1:length(uqnorms))
```

Comment:
I'm a bit surprised there is no way to do the partitioning into unique groups at once (i.e. find unique *and* get the index mapping in one operation).

Reply:
@louisponet and I sat down and found a slightly more efficient way of doing this with dictionaries that also has the benefit of much cleaner code. When you have some time, take a look and let me know what you think.